### DISCUSSION TITLE
CORS

### DISCUSSION THREAD

**Discussion by RJ4100** [uNetworking/uWebSockets.js#577] [General]
Is CORS on uWebSockets enabled by default?

Example:

Domain A: xyz.com has uWS server

Domain B: zyx.com,  creates a wss connection to https://xyz.com/ws1

Is the Access-Control-Allow-Origin header enabled by default?

**Comment by e3dio** [comment]
You write the header if you need it, the browser does not need it set to make cross-origin websocket requests

**Reply by RJ4100** [reply]
Thanks @e3dio!!! So the cross-origin websocket requests are enable by default, right? I would only use these headers if I need them. 

This is exciting! I'm buying Bears! @alexhultman

**Reply by RJ4100** [reply]
@e3dio I would write a cookie to make sure the request is made from a controlled domain; however, if no options handler is required it makes uWebSockets even more appealing.

**Reply by e3dio** [reply]
Beers or Bears ?

![Alaskan_Brown_Bear_062318-9584-Edit_2048x](https://user-images.githubusercontent.com/85405955/125211648-c2eb4800-e25c-11eb-8f73-7fced5e3e5d8.jpg)

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/58995996/125211803-50548980-e2a9-11eb-8a87-eec0109e711a.png)

**Comment by RJ4100** [comment]
@e3dio okay just checked the standard on websockets looks like cors is not implemented. Thanks.

**Comment by RJ4100** [comment]
LOL @alexhultman @e3dio https://imgur.com/a/3fgpp3K

**Comment by RJ4100** [comment]
[https://imgur.com/a/3fgpp3K](url)
